File: frontend/src/data-services/api-services/models/item-search-query-dto.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { ITEM_SORTING_FIELD, LABEL, SORTING_ORDER } from '../../../constants';
import { FilterConditionDto } from './settings/filter-condition-dto';

export interface ItemSortSettingsDTO {
    order: SORTING_ORDER;
    field: ITEM_SORTING_FIELD;
}
export interface ItemSearchQueryDTO {
    ids?: string[];
    originalOrderIds?: string[];
    active?: boolean;
    queueIds?: string[];
    residual?: boolean;
    itemFilters?: FilterConditionDto[];
    lockOwnerIds?: string[];
    holdOwnerIds?: string[];
    labels?: LABEL[];
    tags?: string[];
    labelAuthorIds?: string[];
}

File: backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/model/dto/ItemSearchQueryDTO.java
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

package com.griddynamics.msd365fp.manualreview.queues.model.dto;

import com.griddynamics.msd365fp.manualreview.model.Label;
import com.griddynamics.msd365fp.manualreview.queues.model.ItemFilter;
import com.griddynamics.msd365fp.manualreview.queues.validation.FieldConditionCombination;
import lombok.Data;

import javax.validation.Valid;
import java.util.Set;

@Data
public class ItemSearchQueryDTO {
    private Set<String> ids;
    private Set<String> originalOrderIds;
    private Boolean active;
    private Set<String> queueIds;
    private boolean residual = false;
    private Set<@FieldConditionCombination @Valid ItemFilter> itemFilters;
    private Set<String> lockOwnerIds;
    private Set<String> holdOwnerIds;
    private Set<Label> labels;
    private Set<String> tags;
    private Set<String> labelAuthorIds;
}
